File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-1.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {

        ListNode* currNode1 = list1;
        ListNode* currNode2 = list2;

        ListNode* newListNode = new ListNode(-1);
        ListNode* newListHead = newListNode;

        while(currNode1 != nullptr || currNode2 != nullptr) {

            if (currNode1 == nullptr) {
                newListNode->next = currNode2;
                break;
            } else if (currNode2 == nullptr) {
                newListNode->next = currNode1;
                break;
            } else if (currNode1->val <= currNode2->val){
                newListNode->next = currNode1;
                currNode1 = currNode1->next;
            } else {
                newListNode->next = currNode2;
                currNode2 = currNode2->next;
            }
            
            newListNode = newListNode->next;
        }

        return newListHead->next;
    }
};

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        if not head.next:
            return None

        count = 0
        tmpPtr = head
        while tmpPtr:
            count += 1
            tmpPtr = tmpPtr.next

        removeIndex = count - n
        tmpPtr2 = head

        if removeIndex == 0:
            return head.next

        for i in range(removeIndex - 1):
            tmpPtr2 = tmpPtr2.next
        
        tmpPtr2.next = tmpPtr2.next.next
        return head